analyzers/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs

```diff
@@ -83,25 +84,41 @@ public sealed class SetLocaleForDataTypes : SonarDiagnosticAnalyzer
             c =>
             {
                 var assignmentExpression = (AssignmentExpressionSyntax)c.Node;
+                ProcessExpression(assignmentExpression.Left, c.SemanticModel, symbolsWhereLocaleIsSet);
+            };
 
-                if (GetPropertySymbol(assignmentExpression, c.SemanticModel) is { } propertySymbol
-                    && propertySymbol.Name == "Locale"
-                    && propertySymbol.ContainingType.IsAny(CheckedTypes))
+        private static void ProcessExpression(ExpressionSyntax expression, SemanticModel semanticModel, ISet<ISymbol> symbolsWhereLocaleIsSet)
+        {
+            expression = expression.RemoveParentheses();
+            if (TupleExpressionSyntaxWrapper.IsInstance(expression)
+                && ((TupleExpressionSyntaxWrapper)expression) is var tupleExpression)
```

Comment:
In theory we can change it, but it will be a pain to update it in the future, so suggest not to change it. Only minor things like adding values to SyntaxKindEx for example.

Reply:
@costin-zaharia-sonarsource I was talking with Pavel in our onboarding meeting about this and he was suggesting to ask you about this. Can we change the code-generation logic or is the generated code copy/paste from Tunnel Vision. Especially can we run the script `eng/generate-compiler-code.cmd` mentioned here and change the output from it?

 https://github.com/SonarSource/sonar-dotnet/blob/b556a308bcf88c4bdefa8c47805215f9a58d4b75/analyzers/src/SonarAnalyzer.CFG/ShimLayer/Syntax.xml#L1-L11

Reply:
@costin-zaharia-sonarsource and @pavel-mikula-sonarsource I found the place, where `IsInstance` is generated:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/9c5d064f100186515bcb45aadd629f18bc523a28/StyleCop.Analyzers/StyleCop.Analyzers.CodeGeneration/SyntaxLightupGenerator.cs#L707-L743

sharwell uses a code generator and generates the code based on `Syntax.xml` (see [here](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/9c5d064f100186515bcb45aadd629f18bc523a28/StyleCop.Analyzers/StyleCop.Analyzers.CodeGeneration/SyntaxLightupGenerator.cs#L86-L97)). I don't know whether it is worth integrating the whole generator into our project somehow.

Reply:
Created #5798